apps/src/dropletUtils.js

```diff
@@ -404,7 +417,15 @@ exports.generateAceApiCompleter = function (functionFilter, dropletConfig) {
         callback(null, []);
         return;
       }
-      callback(null, apis);
+      var line = session.getLine(pos.row);
+      var identifier = acUtil.retrievePrecedingIdentifier(line, pos.column);
+      if (pos.column > (identifier.length + 1) &&
+          line[pos.column - identifier.length - 1] === '.') {
```

Comment:
Grabbed @Hamms to help me parse this, and he pointed out that the second condition is correct and checks whether the cursor is inside a symbol that immediately follows a dot.

We eventually tracked down a failing case: a line that begins with `.` will incorrectly take the else case even when the cursor is in such a position.  That's due to an off-by-one error in the first condition; which, incidentally, is unneeded since indexing outside of a string will just return `undefined`.

But that off-by-one error took way too much work to find.  Let's just extract and test a `isCursorInSymbolAfterDot` helper, please?